src/vaadin-pro-grid-inline-editing-mixin.html

```diff
@@ -0,0 +1,239 @@
+        Polymer.Async.timeOut.after(200),
+        this._stopEdit.bind(this));
+    }
+
+    _cancelStopEdit() {
+      // stop edit on outside click will always trigger notify resize.
+      // when tabbing within same row it might not be needed, so cancel
+      if (this._debouncerStopEdit) {
+        this._debouncerStopEdit.cancel();
+        delete this._debouncerStopEdit;
+      }
+    }
+
+    _flushStopEdit() {
+      if (this._debouncerStopEdit) {
+        this._debouncerStopEdit.flush();
+        delete this._debouncerStopEdit;
+      }
+    }
+
+    _enterEditFromEvent(e) {
+      const context = this.getEventContext(e);
+      const column = context.column;
+
+      if (context.item && this._isEditColumn(column)) {
+        if (e.target.localName === 'vaadin-text-field') {
+          return;
+        }
+
+        const path = e.composedPath();
+        const cell = path[path.indexOf(this.$.table) - 3];
+
+        if (!cell || cell.getAttribute('part').indexOf('details-cell') > -1) {
+          return;
+        }
+
+        this._flushStopEdit();
+
+        this._startEdit(cell, column);
+      } else if (this.__edited) {
+        this._stopEdit();
+      }
+    }
+
+    _startEdit(cell, column, shouldResize) {
+      // cancel debouncer enqueued on blur
+      this._cancelStopEdit();
+
+      this._scrollHorizontallyToCell(cell);
+
+      const model = this.__getRowModel(cell.parentElement);
+      this.__edited = {cell, column, model};
+      column._startCellEdit(cell, model);
+
+      this.notifyResize();
+
+      column._setCellEditValue(cell, this._getItemValue(column, model));
+
+      this.addEventListener('item-property-changed', this.__boundItemPropertyChanged);
+    }
+
+    _stopEdit(shouldResize) {
+      if (!this.__edited) {
+        return;
+      }
+      const {cell, column, model} = this.__edited;
+
+      const value = column._getCellEditValue(cell);
+      if (value !== this._getItemValue(column, model)) {
+        this.dispatchEvent(new CustomEvent('item-property-changed', {
+          detail: {
+            index: model.index,
+            item: model.item,
+            path: column.path,
+            value: value
+          },
+          bubbles: true,
+          cancelable: true,
+          composed: true
+        }));
+      }
+
+      column._stopCellEdit(cell, model);
+
+      this.__edited = null;
+
+      this.removeEventListener('item-property-changed', this.__boundItemPropertyChanged);
+
+      this.notifyResize();
+    }
+
+    _switchEditCell(e) {
+      this._cancelStopEdit();
+
+      const cols = this._getEditColumns();
+
+      const {cell, column, model} = this.__edited;
+      const colIndex = cols.indexOf(column);
+      const {index} = model;
+
+      let nextCol = null;
+      let nextIdx = index;
+
+      // enter: move up / down
+      if (e.keyCode === 13) {
+        nextCol = column;
+        nextIdx = e.shiftKey ? index - 1 : index + 1;
+      }
+
+      // tab: move right / left
```

Comment:
Seems that the edit mode is lost on tab press. Worked with a previous revision.

Reply:
Yep, that is by design:

> Hitting tab while editing a cell moves the focus in unedited mode to the cell to the right

It works the same way in Google Spreadsheet, I mentioned this in the morning.